library/X509/ProvidedHook/DbMigration.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/* Icinga Web 2 X.509 Module | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\X509\ProvidedHook;
+
+use Icinga\Application\Hook\MigrationHook;
+use Icinga\Module\X509\Common\Database;
+use Icinga\Module\X509\Model\Schema;
+use ipl\Orm\Query;
+use ipl\Sql\Adapter\Pgsql;
+use ipl\Stdlib\Filter;
+
+class DbMigration extends MigrationHook
+{
+    use Database;
+
+    public function getName(): string
+    {
+        return $this->translate('Icinga Certificate Monitoring');
+    }
+
+    public function providedDescriptions(): array
+    {
+        return [
+            '1.0.0' => $this->translate(
+                'Adjusts the database type of several columns and changes some composed primary keys.'
+            ),
+            '1.1.0' => $this->translate(
+                'Changes the composed x509_target index and x509_certificate valid from/to types to bigint.'
+            ),
+            '1.2.0' => $this->translate(
+                'Changes all timestamp columns to bigint and adjusts enum types of "yes/no" to "n/y".'
+            ),
+            '1.3.0' => $this->translate('Introduces job and job schedules tables.')
+        ];
+    }
+
+    public function getVersion(): string
+    {
+        if ($this->version === null) {
+            $conn = $this->getDb();
+            $schema = Schema::on($conn)->columns('version');
+            if (static::tableExists($conn, $schema->getModel()->getTableName())) {
+                $schema = $schema->first();
+                if ($schema) {
+                    $this->version = $schema->version;
+                } else {
+                    // Schema version table exist, but the user has probably deleted the entry!
+                    $this->version = '1.3.0';
+                }
+            } elseif ($this->getDb() instanceof Pgsql) {
+                // We modified a bunch of timestamp columns to bigint in x509 version 1.2.0.
+                // We have also added Postgres support with x509 version 1.2 and never had an upgrade scripts until now.
+                $this->version = '1.2.0';
+            } else {
+                $columnType = static::getColumnType($conn, 'x509_certificate', 'ctime');
+                if ($columnType !== 'bigint(20) unsigned') {
```

Comment:
The condition `$columnType !== 'bigint(20) unsigned'` only matches in case schema version is  `< 1.2.0`.  Otherwise `$this->version` is still not initialised.